sui_programmability/framework/sources/Governance/Validator.move

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module Sui::Validator {
+    use Std::ASCII::{Self, String};
+    use Std::Option::{Self, Option};
+
+    use Sui::Coin::{Self, Coin};
+    use Sui::ID::{Self, VersionedID};
+    use Sui::SUI::SUI;
+    use Sui::Transfer;
+    use Sui::TxContext::{Self, TxContext};
+
+    friend Sui::Genesis;
+    friend Sui::ValidatorSet;
+
+    /// This happens when someone tries to destroy a Validator object with sui_address
+    /// that doesn't match the sender address.
+    const EADDRESS_MISMATCH: u64 = 0;
+
+    /// This can happen when a validator tries to withdraw stake but doesn't have sufficient balance
+    /// to do so.
+    const EINSUFFICIENT_STAKE: u64 = 1;
+
+    /// This indicates inconsistent internal state, which shouldn't happen and if so is a bug.
+    const EINCONSISTENT_STATE: u64 = 2;
+
+    struct Validator has key, store {
+        id: VersionedID,
+        /// The Sui Address of the validator. This is the sender that created the Validator object,
+        /// and also the address to send validator/coins to during withdraws.
+        sui_address: address,
+        /// A unique human-readable name of this validator.
+        name: String,
+        /// The IP address of the validator.
+        ip_address: u128,
```

Comment:
ip address may not be sufficient as we'll probably need to know port information. Also a Validator may wish to provide a DNS name so that it doesn't require a fixed static ip address.

Reply:
Let me make this a vector<u8> so that it could contain anything